From: Matthias Brugger <mbrugger@suse.com>
Date: Thu 14 Apr 20:35:44 CEST 2016
Subject: net: thunderx: Don't use mdio
Patch-mainline: Never, fixes a local compilation error

Kernel v4.4 does not have commit 
e5a03bfd873c ("phy: Add an mdio_device structure").
This patch fixes a compilation error and we don't have to touch the mailine patch.

--- a/drivers/net/ethernet/cavium/thunder/thunder_bgx.c
+++ b/drivers/net/ethernet/cavium/thunder/thunder_bgx.c
@@ -1023,7 +1023,7 @@ defer:
 	 */
 	while (lmac) {
 		if (bgx->lmac[lmac].phydev) {
-			put_device(&bgx->lmac[lmac].phydev->mdio.dev);
+			put_device(&bgx->lmac[lmac].phydev->dev);
 			bgx->lmac[lmac].phydev = NULL;
 		}
 		lmac--;
